samcli/lib/utils/git_repo.py

```diff
@@ -162,3 +175,17 @@ def _persist_local_repo(temp_path: str, dest_dir: Path, dest_name: str, replace_
                 f"Check that you have permissions to create/delete files in {dest_dir} directory "
                 "or file an issue at https://github.com/aws/aws-sam-cli/issues"
             ) from ex
+
+    @staticmethod
+    def _checkout_commit(repo_dir: str, commit: str, git_executable: str):
```

Comment:
Does this function need to be protected?

Reply:
We don't use it in other places, keep it private now